ablySpec/RealtimeClientChannel.swift

```diff
@@ -181,6 +181,25 @@ class RealtimeClientChannel: QuickSpec {
                     expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Failed), timeout: testTimeout)
                 }
 
+                // RTL4f
+                pending("should transition the channel state to FAILED if ATTACHED ProtocolMessage is not received") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let client = ARTRealtime(options: options)
+                    client.setTransportClass(TestProxyTransport.self)
+                    client.connect()
+                    defer { client.close() }
+
+                    expect(client.connection().state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
+                    let transport = client.transport as! TestProxyTransport
+                    transport.actionsIgnored += [.Attached]
+
+                    let channel = client.channel("test")
+                    channel.attach()
```

Comment:
See http://docs.ably.io/client-lib-development-guide/features/#DF1b, it's not a documented API, but we've made this configurable in our client libraries and left it undocumented for exactly this reason